optimum/amd/ryzenai/models/yolov3/image_processing_yolov3.py

```diff
@@ -0,0 +1,180 @@
+# Copyright 2023 The HuggingFace Team. All rights reserved.
+# Licensed under the MIT License.
+
+from typing import Dict, List, Optional, Tuple, Union
+
+import numpy as np
+import torch
+
+from transformers.image_processing_utils import BaseImageProcessor, BatchFeature
+from transformers.image_transforms import (
+    rescale,
+)
+from transformers.image_utils import (
+    ChannelDimension,
+    ImageInput,
+    infer_channel_dimension_format,
+    make_list_of_images,
+    to_numpy_array,
+)
+from transformers.utils import TensorType
+
+from ..detection_utils import non_max_suppression, scale_coords
+from ..image_transforms import letterbox_image
+
+
+def make_grid(anchor, nx=20, ny=20):
+    d = anchor.device
+    t = anchor.dtype
+
+    shape = 1, 1, ny, nx, 2
+
+    y, x = torch.arange(ny, device=d, dtype=t), torch.arange(nx, device=d, dtype=t)
+    yv, xv = torch.meshgrid(y, x, indexing="ij")
+
+    grid = torch.stack((xv, yv), 2).expand(shape)
+    anchor_grid = (anchor).reshape(3, 2).view((1, len(anchor) // 2, 1, 1, 2))
+
+    return grid, anchor_grid
+
+
+def postprocess(inputs, anchors, num_classes=80, stride=[8, 16, 32], shapes=[80, 40, 20]):
+    nl = len(anchors)
+    no = num_classes + 5
+
+    outputs = []
+    for i in range(nl):
+        bs, _, ny, nx = inputs[i].shape
+        grid, anchor_grid = make_grid(anchors[2 - i], nx, ny)
+
+        inputs[i] = inputs[i].view(bs, nl, no, ny, nx).permute(0, 1, 3, 4, 2).contiguous()
+
+        xy = (torch.sigmoid(inputs[i][..., 0:2]) + grid) * stride[2 - i]
+        wh = (torch.exp(inputs[i][..., 2:4])) * anchor_grid
+
+        conf = torch.sigmoid_(inputs[i][..., 4:])
+        y = torch.cat((xy, wh, conf), -1)
+        outputs.append(y.view(bs, -1, no))
+    return torch.cat(outputs, 1)
+
+
+class YoloV3ImageProcessor(BaseImageProcessor):
```

Comment:
Why do we need this here? In optimum-amd? Is it not already in transformers?

Reply:
Most of the CV models (including yolo and other image2image models) from Ryzen `model_zoo` are not present in transformers. So added these here!

Reply:
Unfortunately, none of the HF libraries have these models or similar detection utilities. So, I had to include them here. However, I did receive help from the sample code provided by AMD.